common/app-starters-security-common/src/test/java/org/springframework/cloud/stream/app/security/common/AbstractSecurityCommonTests.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright 2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.cloud.stream.app.security.common;
+
+import org.junit.runner.RunWith;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.web.client.TestRestTemplate;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+
+/**
+ * @author Christian Tzolov
+ * @author Artem Bilan
+ */
+@RunWith(SpringJUnit4ClassRunner.class)
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+public abstract class AbstractSecurityCommonTests {
+
+	@Autowired
+	protected TestRestTemplate restTemplate;
+
+	@SpringBootApplication
+	public static class AutoConfigurationApplication {
```

Comment:
This is not good name for the test configuration, but let it be.

Another comment (not critical) about that `@SpringBootApplication`. It's not good practice to let test configuration to scan all your classpath.

So, better way to isolate the scope only with the test class is to use this combination:
```
@SpringBootConfiguration
@EnableAutoConfiguration
```

Reply:
+1

Reply:
+1

Reply:
Christian, I have a lot of spam today from you.
All of that is just because you agreed with me.

How about to agree silently and just push a fix? 😄 

Much less emails to read.

Just my feeling about all this "+1" 😸